### ISSUE TITLE
Invalid argument error: while implementing custom loss

### ISSUE THREAD

**Issue by shivam05011996** [keras-team/keras#10408]
I am getting a InvalidArgumentError while implementing custom loss function as-
```
InvalidArgumentError (see above for traceback): Incompatible shapes: [5] vs. [32]
         [[Node: training/Adam/gradients/loss/dense_11_loss/mul_grad/BroadcastGradientArgs = BroadcastGradientArgs[T=DT_INT32, _class=["loc:@training/Adam/gradients/loss/dense_11_loss/mul_grad/Reshape"], 
_device="/job:localhost/replica:0/task:0/device:GPU:0"](training/Adam/gradients/loss/dense_11_loss/mul_grad/Shape, training/Adam/gradients/loss/dense_11_loss/mul_grad/Shape_1)]]
         [[Node: loss/mul/_199 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnat
ion=1, tensor_name="edge_1145_loss/mul", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

```
My custom loss function - 
```
def _euclidean_dis_loss(self, y_true, y_pred):
    return K.sqrt(K.sum(K.square(y_pred - y_true), axis=0))
```
Number of dimension in a vector = 5
Batch Size = 32
But when i change the axis to -1, it works perfectly.

Any help or comment appreciated.

**Comment by brge17** [comment]
Several issues:

1. Take it outside the class the ``self`` attribute isn't used (nor should it be).

2. As noted in https://github.com/keras-team/keras/pull/10400, your loss scales with batch size (which is bad practice) because of the ``K.sum``.

3. As noted in https://github.com/keras-team/keras/pull/10400, you should just use MSE.

4. ``axis=0`` is the batch dimension, it should be along the last dimension ``axis=1``. Look at ``MSE`` implementation or any other loss function for that matter (https://github.com/keras-team/keras/blob/master/keras/losses.py).

**Comment by shivam05011996** [comment]
@brge17 This problem was sloved, with on more argument , keep_dims=True,
The rest of the points you mentioned, I agree with them all, using just MSE, has resulted me better loss numbers.
